src/sentry/models/event.py

```diff
@@ -633,6 +619,16 @@ class Meta:
 
     __repr__ = sane_repr("project_id", "group_id")
 
+    def __init__(self, *args, **kwargs):
```

Comment:
I think this allowing this to take the same argument in two different types is going to be very confusing.
I would add factory methods, one for NodeData and ther other for the other type you allow, and please add a docstring to each explaining when to use them. 

But taking a step back, can we converge into one type only ?

Reply:
The `clear_node_id` method attempts to make this a bit clearer